openfe/protocols/restraint_utils/geometry/boresch.py

```diff
@@ -0,0 +1,870 @@
+
+    # 0. Get the ligand RMSF
+    rmsf = get_local_rmsf(ligand_ag)
+    u.trajectory[-1]  # forward to the last frame
+
+    # 1. Get the pool of atoms to work with
+    atom_pool = _get_atom_pool(rdmol, rmsf, rmsf_cutoff)
+
+    if atom_pool is None:
+        # We don't have enough atoms so we raise an error
+        errmsg = "No suitable ligand atoms were found for the restraint"
+        raise ValueError(errmsg)
+
+    # 2. Get the central atom
+    center = get_central_atom_idx(rdmol)
+
+    # 3. Sort the atom pool based on their distance from the center
+    sorted_atom_pool = _sort_by_distance_from_atom(rdmol, center, atom_pool)
+
+    # 4. Get a list of probable angles
+    angles_list = []
+    for atom in sorted_atom_pool:
+        angles = _bonded_angles_from_pool(rdmol, atom, sorted_atom_pool)
+        for angle in angles:
+            # Check that the angle is at least not collinear
+            angle_ag = ligand_ag.atoms[list(angle)]
+            if not is_collinear(ligand_ag.positions, angle, u.dimensions):
+                angles_list.append(
+                    (
+                        angle_ag.atoms[0].ix,
+                        angle_ag.atoms[1].ix,
+                        angle_ag.atoms[2].ix
+                    )
+                )
+
+    return angles_list
+
+
+def find_host_atom_candidates(
+    topology: Union[str, pathlib.Path, openmm.app.Topology],
+    trajectory: Union[str, pathlib.Path, npt.NDArray],
+    host_idxs: list[int],
+    l1_idx: int,
+    host_selection: str,
+    dssp_filter: bool = False,
+    rmsf_cutoff: unit.Quantity = 0.1 * unit.nanometer,
+    min_distance: unit.Quantity = 1 * unit.nanometer,
+    max_distance: unit.Quantity = 3 * unit.nanometer,
+) -> npt.NDArray:
+    """
+    Get a list of suitable host atoms.
+
+    Parameters
+    ----------
+    topology : Union[str, openmm.app.Topology]
+      The topology of the system.
+    trajectory : Union[str, pathlib.Path, npt.NDArray]
+      The system's coordinate trajectory.
+    host_idxs : list[int]
+      A list of the host indices in the system topology.
+    l1_idx : int
+      The index of the proposed l1 binding atom.
+    host_selection : str
+      An MDAnalysis selection string to fileter the host by.
+    dssp_filter : bool
+      Whether or not to apply a DSSP filter on the host selection.
+    rmsf_cutoff : uni.Quantity
+      The maximum RMSF value allowwed for any candidate host atom.
+    min_distance : unit.Quantity
+      The minimum search distance around l1 for suitable candidate atoms.
+    max_distance : unit.Quantity
+      The maximum search distance around l1 for suitable candidate atoms.
+
+    Return
+    ------
+    NDArray
+      Array of host atom indexes
+    """
+    u = mda.Universe(
+        topology,
+        trajectory,
+        format=_get_mda_coord_format(trajectory),
+        topology_format=_get_mda_topology_format(topology),
+    )
+
+    host_ag1 = u.atoms[host_idxs]
```

Comment:
Maybe find better words for this (or maybe actually explain what the variable means!)